### PR TITLE
Add support to bazel.

### PR DESCRIPTION
**Pull request by hcoona**
Add support to [Bazel](https://bazel.build/) build system.

Tested on the following platforms:

1. Windows + VS2019
2. Ubuntu 18.04.2 + G++ 7.4.0
3. Ubuntu 18.04.2 + Clang++ 8.0.1

with [bazelisk](https://github.com/bazelbuild/bazelisk):

* Bazelisk version: v0.0.8
* Build label: 0.27.1

### CODE REVIEW

**Comment by nitronoid** [comment]
+1 for this

**Comment by nitronoid** [comment]
To get the tests to work, I had to manually set the c++ standard in my .bazel rc. 
i.e. 
```bash
bash$ cat .bazelrc
build --cxxopt='-std=c++17'
```
@hcoona Maybe you could add this to your branch?

**Comment by hcoona** [comment]
@nitronoid We need to tests GSL against different c++ standard versions: c++11, c++14, c++17. As a result, I don't want to attach the arguments in `.bazelrc` file. Please reference the `.travis.yml` for more details.

**Comment by hcoona** [comment]
@JordanMaples Would you mind to take a look at this? It's a very simple change & no core code change.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+root = true
```
Comment (hanst99) [inline .editorconfig]: The `.editorconfig` file seems unnecessary?

File: include/gsl/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,19 @@
+cc_library(
```
Comment (hanst99) [inline include/gsl/BUILD.bazel]: Why is this file in the `include` directory?

Comment (hcoona) [reply include/gsl/BUILD.bazel:1]: It's an convention from bazel that regard each folder as a package. The build file is for the contents in this package.

**Pull request closed without merging**